dartagnan/src/main/java/com/dat3m/dartagnan/encoding/SymmetryEncoder.java

```diff
@@ -81,59 +90,56 @@ public static SymmetryEncoder withContext(EncodingContext context, Wmm memoryMod
     public void initializeEncoding(SolverContext context) { }
 
     public BooleanFormula encodeFullSymmetry() {
-        BooleanFormulaManager bmgr = context.getBooleanFormulaManager();
+        final BooleanFormulaManager bmgr = context.getBooleanFormulaManager();
         BooleanFormula enc = bmgr.makeTrue();
         if (rel == null) {
             return enc;
         }
-
         for (EquivalenceClass<Thread> symmClass : symm.getNonTrivialClasses()) {
-            enc = bmgr.and(enc, encodeSymmetryClass(symmClass));
+            enc = bmgr.and(enc,  encodeSymmetryClass(rel, symmClass));
         }
         return enc;
     }
 
-    public BooleanFormula encodeSymmetryClass(EquivalenceClass<Thread> symmClass) {
-        SolverContext ctx = context.getSolverContext();
-        BooleanFormulaManager bmgr = ctx.getFormulaManager().getBooleanFormulaManager();
-        BooleanFormula enc = bmgr.makeTrue();
-        if (rel == null || symmClass.getEquivalence() != symm) {
-            return enc;
-        }
-
-        Thread rep = symmClass.getRepresentative();
-        List<Thread> symmThreads = new ArrayList<>(symmClass);
+    private BooleanFormula encodeSymmetryClass(Relation rel, EquivalenceClass<Thread> symmClass) {
+        Preconditions.checkArgument(symmClass.getEquivalence() == symm,
+                "Symmetry class belongs to unexpected symmetry relation.");
+        Preconditions.checkState(rel != null, "Relation to break symmetry on is NULL");
```

Comment:
relation -> concept (or whatever used in the description)

Reply:
This is private code which does take a relation as input. The message is not intended to be seen by users.
But its good you pointed it out, because it should not be a state condition anymore :)